feat(language_server): provide commands / code actions for unopened files

Sysix opened the pull request:
Related https://github.com/oxc-project/oxc-intellij-plugin/pull/177

When the client request code actions or commands for a file. This file must not be open to be requested.
In the main, we are just looking into our `diagnostics_map`, which will be filled only on open/change/close of the text document.

When the client request now a code action or command for a file, we will lint them only (if not already done) and avoid filling the `diagnostic_map`.

Sysix commented:
* **#10815** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10815?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10815?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=10815).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

camc314 commented:
### Merge activity

* **May 6, 4:38 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Sysix committed:
feat(language_server): provide commands / code actions for unopened files (#10815)

Related https://github.com/oxc-project/oxc-intellij-plugin/pull/177

When the client request code actions or commands for a file. This file must not be open to be requested.
In the main, we are just looking into our `diagnostics_map`, which will be filled only on open/change/close of the text document.

When the client request now a code action or command for a file, we will lint them only (if not already done) and avoid filling the `diagnostic_map`.
